internal/testutil/testutil.go

```diff
@@ -0,0 +1,62 @@
+package testutil
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type MockCmd struct {
+	binDir string
+	name   string
+}
+
+func MockCommand(t *testing.T, name string, script string) *MockCmd {
+	mockCmd := &MockCmd{
+		binDir: t.TempDir(),
+		name:   name,
+	}
+
+	fullScript := `#!/bin/bash -e
+for arg in "$@"; do
+   echo -e -n "$arg\x0" >> "$0".run
+done
+echo >> "$0".run
+` + script
+
+	t.Setenv("PATH", mockCmd.binDir+":"+os.Getenv("PATH"))
+	err := os.WriteFile(filepath.Join(mockCmd.binDir, name), []byte(fullScript), 0755)
+	require.NoError(t, err)
+
+	return mockCmd
+}
+
+func (mc *MockCmd) Path() string {
+	return filepath.Join(mc.binDir, mc.name)
+}
+
+func (mc *MockCmd) Restore() error {
```

Comment:
`t.TempDir` gets removed automatically, so we should not need `Restore` at all. See https://pkg.go.dev/testing#T.TempDir

Reply:
Thanks, good point! The way we use it currently I guess we could remove it (I'm tempted :) - in the general case it might be useful though if one would want to "unmock" a command during test runs (which is a rare thing to do I guess).